### PR TITLE
Added Auth.php - implementation of Guard

### PR DESCRIPTION
**Pull request by JaffParker**
Created an implementation of Illuminate\Contracts\Auth\Guard for seamless replacement of standard Laravel Auth with JWTAuth.

### CODE REVIEW

**Comment by crynobone** [comment]
Would actually love to see a new Auth "guard" (driver) based on 5.2 new structure.

**Comment by JaffParker** [comment]
@crynobone I'll check it out

**Comment by jonagoldman** [comment]
+1 Specially now that 5.2 is officially released.

File: src/Auth.php [hunk 1]
```diff
@@ -0,0 +1,171 @@
+<?php namespace Tymon\JWTAuth;
+
+use Illuminate\Contracts\Auth\Authenticatable;
+use Illuminate\Contracts\Auth\Guard;
+use Illuminate\Http\Request;
+use Koodzo\Models\Db\User;
```
Comment (crynobone) [inline src/Auth.php]: I don't think this is a generic `User` model.

**Comment by tymondesigns** [comment]
closing as I will be looking to add proper support for the Auth Guard stuff that was added in 5.2

**Pull request closed without merging**